A66-otel-stats.md

```diff
@@ -0,0 +1,566 @@
+  // to record the target attribute on client or to replace it with "other".
+  // This helps reduce the cardinality on metrics in cases where many channels
+  // are created with different targets in the same binary (which might happen
+  // for example, if the channel target string uses IP addresses directly).
+  OpenTelemetryPluginBuilder& SetTargetAttributeFilter(
+      absl::AnyInvocable<bool(absl::string_view /*target*/) const>
+          target_attribute_filter);
+  // If set, \a generic_method_attribute_filter is called per call with a
+  // generic method type to decide whether to record the method name or to
+  // replace it with "other". Non-generic or pre-registered methods remain
+  // unaffected. If not set, by default, generic method names are replaced with
+  // "other" when recording metrics.
+  OpenTelemetryPluginBuilder& SetGenericMethodAttributeFilter(
+      absl::AnyInvocable<bool(absl::string_view /*generic_method*/) const>
+          generic_method_attribute_filter);
+  // Registers a global plugin that acts on all channels and servers running on
+  // the process.
+  // The most common way to use this API is -
+  //
+  // OpenTelemetryPluginBuilder().SetMeterProvider(provider)
+  //    .BuildAndRegisterGlobal();
+  //
+  // The set of instruments available are -
+  // grpc.client.attempt.started
+  // grpc.client.attempt.duration
+  // grpc.client.attempt.sent_total_compressed_message_size
+  // grpc.client.attempt.rcvd_total_compressed_message_size
+  // grpc.server.call.started
+  // grpc.server.call.duration
+  // grpc.server.call.sent_total_compressed_message_size
+  // grpc.server.call.rcvd_total_compressed_message_size
+  void BuildAndRegisterGlobal();
+};
+
+```
+
+In the future, additional API might be provided to allow registering the plugin
+for a particular channel or server builder.
+
+#### Java
+
+```java
+public static class OpenTelemetryModuleBuilder {
+    /**
+     * OpenTelemetry instance is used to configure metrics settings.
+     *
+     * Sample
+     *    SdkMeterProvider sdkMeterProvider = SdkMeterProvider.builder()
+     *         .registerMetricReader(
+     *             PeriodicMetricReader.builder(
+     *                OtlpGrpcMetricExporter.builder().build()).build())
+     *         .build();
+     *
+     *     OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+     *         .setMeterProvider(sdkMeterProvider)
+     *         .build();
+     *
+     * If MeterProvider is not configured, no-op meterProvider will be used by default.
+     * It provides meters which do not record or emit.
+     */
+    public OpenTelemetryModuleBuilder openTelemetry(OpenTelemetry openTelemetry);
+
+    /* If targetFilter is set, and returns true for a target, target is recorded as is. Records "other" on false.
+    If targetFilter is not set, target is recorded as is. */
+    public OpenTelemetryBuilder targetFilter(Predicate<String> targetFilter);
+
+    public OpenTelemetryModule build();
```

Comment:
We won't want simple "build" to install globally (although it's not clear if that is the intention here). I'm fine with fixing up this API afterward, as long as others are. The sketch is fair, but the spellings will probably morph a bit.

(Similarly, I don't think we'll use the method name `openTelemetry()` above.)

Reply:
sounds fine to me

Reply:
I avoided using global in the build method, because [`buildAndRegisterGlobal`](https://javadoc.io/doc/io.opentelemetry/opentelemetry-sdk/latest/io/opentelemetry/sdk/OpenTelemetrySdkBuilder.html#buildAndRegisterGlobal()) has different meaning wrt OpenTelemetry java. In OpenTelemetry it also sets the sdk instance invoking the instance as Global instance which I didn't want to inferred here as well. 

I named method as `openTelemetry()` since we are taking an openTelemetry instance. How does `openTelemetrySdk()` or `sdk()` sound?